src/sage/rings/valuation/valuation_space.py

```diff
@@ -0,0 +1,1573 @@
+                True
+                sage: v.uniformizer()
+                Traceback (most recent call last):
+                ...
+                ValueError: Trivial valuations do not define a uniformizing element
+                
+            """
+
+        @cached_method
+        def value_group(self):
+            r"""
+            Return the value group of this discrete pseudo-valuation, the
+            discrete additive subgroup of the rational numbers which is
+            generated by the valuation of the :meth:`uniformizer`.
+
+            EXAMPLES::
+
+                sage: QQ.valuation(2).value_group()
+                Additive Abelian Group generated by 1
+
+            A pseudo-valuation that is `\infty` everywhere, does not have a
+            value group::
+
+                sage: from sage.rings.valuation.valuation_space import DiscretePseudoValuationSpace
+                sage: v = DiscretePseudoValuationSpace(QQ).an_element()
+                sage: v.value_group()
+                Traceback (most recent call last):
+                ...
+                ValueError: The trivial pseudo-valuation that is infinity everywhere does not have a value group.
+
+            """
+            from value_group import DiscreteValueGroup
+            return DiscreteValueGroup(self(self.uniformizer()))
+
+        def value_semigroup(self):
+            r"""
+            Return the value semigroup of this discrete pseudo-valuation, the
+            additive subsemigroup of the rational numbers which is generated by
+            the valuations of the elements in :meth:`domain`.
+
+            EXAMPLES:
+
+            Most commonly, in particular over fields, the semigroup is the
+            group generated by the valuation of the uniformizer::
+
+                sage: G = QQ.valuation(2).value_semigroup(); G
+                Additive Abelian Semigroup generated by -1, 1
+                sage: G in AdditiveMagmas().AdditiveAssociative().AdditiveUnital().AdditiveInverse()
+                True
+
+            If the domain is a discrete valuation ring, then the semigroup
+            consists of the positive elements of the :meth:`value_group`::
+
+                sage: Zp(2).valuation().value_semigroup()
+                Additive Abelian Semigroup generated by 1
+
+            The semigroup can have a more complicated structure when the
+            uniformizer is not in the domain::
+
+                sage: v = ZZ.valuation(2)
+                sage: R.<x> = ZZ[]
+                sage: w = GaussValuation(R, v)
+                sage: u = w.augmentation(x, 5/3)
+                sage: u.value_semigroup()
+                Additive Abelian Semigroup generated by 1, 5/3
+
+            """
+            from sage.categories.fields import Fields
+            if self.domain() in Fields():
+                from value_group import DiscreteValueSemigroup
+                return DiscreteValueSemigroup([]) + self.value_group()
```

Comment:
This is an easy way to create the semigroup generated by the group. I added a comment in the code.